connect/resource/automation.py

```diff
@@ -43,17 +43,17 @@ def process_request(self, request):
     @function_log
     def approve(self, pk, data):
         # type: (str, dict) -> str
-        return self.api.post(path=pk + '/approve/', data=data if data else {})
+        return self.api.post(path=pk + '/approve/', data=json.dumps(data))
```

Comment:
Probably, better to put that functionality to into 
api.post/api.put 
if data is not instance of None/string/bytes then serialize it with json.dumps?

Reply:
That would simplify use of ApiClient, and would work for request against Connect, where the body is JSON-encoded, but I want ApiClient to be usable by connectors to call vendor APIs (as we talked on the meeting about logging), and in that case, in some situations we might want to pass the dictionary directly to requests so it can be form-encoded instead of converted to JSON.

If we want to simplify this, we could use a different approach to this: Instead of encoding JSON as a string and passing it to the data parameter, requests supports a json argument since version 2.4.2 (and we are using 2.21.0, so we are good to go). If you put a dictionary here, is is encoded to a JSON string automatically. This would mean using the json arguments instead of data in most places, and would have a concise syntax.

The only problem would be providing a standard mechanism to trivially create a dictionary from any model so we can put it in the json argument. This could be very simply done by adding a "json" property to BaseModel.

I am going to make these changes so you can take a look at it.

Reply:
my 5 cents: Even that your goal is nice, i don't think is a real key requierement. People is used to certain libraries, trying to enforce use of X of vendor's api is bit unrealistic. SDK must cover API of Connect rest is limiting scope.